### PR TITLE
refactor: reuse open file in storage interface

### PR DESCRIPTION
**Pull request by ferhatelmas**
## Description
Implement sync open file via reusing no sync open file.

## Motivation and Context
To prevent divergence in the same implementation and code reuse/dry, related to #16958.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/xl-storage.go [hunk 1]
```diff
@@ -1672,32 +1672,7 @@ func (s *xlStorage) openFileDirect(path string, mode int) (f *os.File, err error
...
-			return nil, errFaultyDisk
-		case isSysErrTooManyFiles(err):
-			return nil, errTooManyOpenFiles
-		default:
-			return nil, err
-		}
-	}
-
-	return w, nil
+	return s.openFileNoSync(filePath, mode|writeMode)
```
Comment (harshavardhana) [inline cmd/xl-storage.go]: There is a reason why we want sync to be sync and not `nosync` - This code is not correct @ferhatelmas

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/xl-storage.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/xl-storage.go]: they have a different function signature, we should change `s.openFileNoSync` to be `s.openFile`

Comment (ferhatelmas) [reply cmd/xl-storage.go]: so next step is to rename `openFileNoSync` to `openFile` and keep this change (call within sync with right mode) right?

Comment (harshavardhana) [reply cmd/xl-storage.go]: yes @ferhatelmas

Comment (ferhatelmas) [reply cmd/xl-storage.go]: thanks, updated

**Review by harshavardhana — APPROVED** [review]

**Review by klauspost — APPROVED** [review]

**Pull request merged**